### ISSUE TITLE
Make add_school_work_participation() function use data from official data sources like ABS

### ISSUE THREAD

**Issue by njtierney** [njtierney/conmat#15]
e.g., https://github.com/njtierney/conmat/blob/master/R/add_school_work_participation.R

**Comment by njtierney** [comment]
So for example, in these lines:

https://github.com/njtierney/conmat/blob/master/R/add_school_work_participation.R#L25-L34

Instead of the numbers, 0.5, 1, etc., we would derive the fraction of people in those age groups in school inside the function and then use those numbers.

We can start by internally using the abs education data, but as seen in https://github.com/njtierney/conmat/issues/14 we need to make sure we can trust the education data.

Perhaps in the future then we can extend this so people can use their own dataset that isn't the ABS or something, but for the time being using the ABS data is great.

**Comment by njtierney** [comment]
Discussing with Aarathy some notes on this:

- [ ] Calculate proportion summary similar to these numbers (2-4, 5-16, etc) https://github.com/njtierney/conmat/blob/713f08bed70715bc6fa9ee0030e245acc3b551e5/R/add_school_work_participation.R#L24-L47) 
- [ ] Then compare two approaches using `complete(age = 0:100)` vs `complete(0:24)` and see if those numbers are different

**Comment by njtierney** [comment]
Worth noting is the education referring to only primary and secondary education

Do we want to also include tertiary education data? 

Also would be good to confirm what "ungraded primary/secondary" is exactly.

@goldingn what are your thoughts on making this more extensible, in terms of allowing users to provide different datasets to then estimate school and work fraction, so that they would end up with different proportions instead of:

```r
   # preschool 
   .x %in% 2:4 ~ 0.5, 
   # compulsory education 
   .x %in% 5:16 ~ 1, 
   # voluntary education 
   .x %in% 17:18 ~ 0.5, 
   # university 
   .x %in% 19:25 ~ 0.1, 
   # other 
   TRUE ~ 0.05 
```

**Comment by aarathybabu97** [comment]
#### Compare work and school fractions with fractions in conmat

Data sourced from the 2016 Census - Employment,  Income, and Education obtained through TableBuilder.  ABS  has randomly adjusted the data to avoid the release of confidential data.


- Work fraction 

Uses the ABS state-wise data on the number of employed persons of a particular age (0-115) and the total population of that age. 

```r 
data_abs_state_work
#> # A tibble: 1,044 × 6
#>     year state   age employed_population total_population proportion
#>    <dbl> <chr> <dbl>               <dbl>            <dbl>      <dbl>
#>  1  2016 NSW       0                   0            87708          0
#>  2  2016 NSW       1                   0            92876          0
#>  3  2016 NSW       2                   0            93584          0
#>  4  2016 NSW       3                   0            95179          0
#>  5  2016 NSW       4                   0            95791          0
#>  6  2016 NSW       5                   0            95216          0
#>  7  2016 NSW       6                   0            96479          0
#>  8  2016 NSW       7                   0            95142          0
#>  9  2016 NSW       8                   0            95833          0
#> 10  2016 NSW       9                   0            95516          0
#> # … with 1,034 more rows
```


```r
work_fraction_comparison_table
#> # A tibble: 5 × 3
#>   age_group conmat_work_prop work_fraction
#>   <chr>                <dbl>         <dbl>
#> 1 12-19                 0.2         0.238 
#> 2 20-24                 0.7         0.636 
#> 3 25-60                 1           0.724 
#> 4 61-65                 0.7         0.446 
#> 5 Other                 0.05        0.0528

````

- School fraction 

Uses the ABS state-wise data on the number of students (full-time, part-time &  persons whose institution (TYPP) is stated, but not the full-time/part-time status (STUP))  of a particular age (0-115) and the total population of that age. 

```r
data_abs_state_education
#> # A tibble: 1,044 × 6
#>     year state   age population_educated total_population proportion
#>    <dbl> <chr> <dbl>               <dbl>            <dbl>      <dbl>
#>  1  2016 NSW       0                   0            87708      0    
#>  2  2016 NSW       1                   0            92876      0    
#>  3  2016 NSW       2                   0            93584      0    
#>  4  2016 NSW       3               45977            95179      0.483
#>  5  2016 NSW       4               66858            95791      0.698
#>  6  2016 NSW       5               84229            95216      0.885
#>  7  2016 NSW       6               91479            96479      0.948
#>  8  2016 NSW       7               90412            95142      0.950
#>  9  2016 NSW       8               91079            95833      0.950
#> 10  2016 NSW       9               90643            95516      0.949
#> # … with 1,034 more rows
```


```r
school_comparison_table
#> # A tibble: 5 × 3
#>   age_group conmat_prop   prop
#>   <chr>           <dbl>  <dbl>
#> 1 2-4              0.5  0.342 
#> 2 5-16             1    0.940 
#> 3 17-18            0.5  0.737 
#> 4 19-25            0.1  0.390 
#> 5 Other            0.05 0.0506
```

**Comment by njtierney** [comment]
closed by #153
